Street Trees/STConfirmationPageViewController.swift

```diff
@@ -0,0 +1,179 @@
+    }
+    
+    func itemsForSection(atIndex anIndex: Int) -> Int {
+        return self.dataSource[anIndex].count
+    }
+    
+    func item(forRow aRow: Int, inSection aSection: Int) -> STConfirmationDetailRows {
+        return self.dataSource[aSection][aRow]
+    }
+    
+    init(dataSource:[[STConfirmationDetailRows]]) {
+        self.dataSource = dataSource
+    }
+}
+
+//**********************************************************************************************************************
+// MARK: - Protocol
+
+protocol STConfirmationPageViewControllerDelegate: NSObjectProtocol
+{
+    func confirmationFormViewController(form: STConfirmationPageViewController, didCompleteWithWufooForm anWufooForm: STTKWufooForm)
+}
+
+//**********************************************************************************************************************
+
+class STConfirmationPageViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, TreeDescription, Address, Contact
+{
+    
+    var treeDescription:STPKTreeDescription?
+    var address: STTKStreetAddress?
+    var contact: STTKContact?
+ 
+    
+    var dataSource: DataSource = DataSource(dataSource: [[.TreeName], [.Address, .CityStateZip], [.UserName, .Phone, .email]])
+    
+    weak var  delegate:STConfirmationPageViewControllerDelegate?
+
+    // MARK: - Table View Methods:
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int
+    {
+        return dataSource.sections
+    }
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
+    {
+        return dataSource.itemsForSection(atIndex: section)
+    }
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell
+    {
+        let cell = tableView.dequeueReusableCellWithIdentifier("emptyCell", forIndexPath: indexPath)
+
+        let kindOfRow = dataSource.item(forRow: indexPath.row, inSection: indexPath.section)
+        
+        var textLabel:String? = nil
+        var imageView:UIImage? = nil
+        
+        switch kindOfRow
+        {
+        case .TreeName:
+            textLabel = treeDescription?.name ?? "Error retrieving tree name"
+            imageView = treeDescription?.image() ?? UIImage(named: "blankMap")
+        case .Address:
+            textLabel = (address?.streetAddress)!  + " " +  (address?.secondaryAddress)! ?? "Error retrieving address"
+        case .CityStateZip:
+            textLabel = "\(address?.city), \(address?.state) \(String((address?.zipCode)!))" ?? "Error retrieving location"
+        case .UserName:
+            textLabel = (contact?.name)! ?? "Error retrieving name"
+        case .Phone:
+            textLabel = (contact?.phoneNumber)! ?? "Error retrieving phone number"
+        case .email:
+            textLabel = (contact?.email)! ?? "Error retrieving email"
+        }
+        
+        cell.textLabel?.text = textLabel
+        cell.imageView?.image = imageView
```

Comment:
imageView is only ever added once. Move this cell.imageView?.image to the .TreeName case.